packages/jupyter-ai/src/hooks/use-copy.ts

```diff
@@ -38,15 +47,19 @@ export type UseCopyReturn = {
 const DEFAULT_LABELS_BY_COPY_STATUS: Record<CopyStatus, string> = {
   [CopyStatus.None]: 'Copy to clipboard',
   [CopyStatus.Copying]: 'Copying…',
-  [CopyStatus.Copied]: 'Copied!'
+  [CopyStatus.Copied]: 'Copied!',
+  [CopyStatus.Disabled]: 'Copy to clipboard disabled in insecure context'
```

Comment:
We should provide more helpful information to users in the tooltip, since many users will not know what an insecure context is.

```suggestion
  [CopyStatus.Disabled]: 'The clipboard is disabled by your browser because you are accessing this page outside of a secure context. To enable the clipboard, you must access this page through HTTPS after ensuring the server is configured to accept HTTPS connections.'
```

You may have to adjust the CSS set by the `sx` prop in MUI React Elements to get this to render in a readable way.